client/client.go

```diff
@@ -1247,6 +1289,207 @@ func (c *Client) runAllocs(update *allocUpdates) {
 	}
 }
 
+// blockForRemoteAlloc blocks until the previous allocation of an allocation has
+// been terminated and migrates the snapshot data
+func (c *Client) blockForRemoteAlloc(alloc *structs.Allocation) {
```

Comment:
I would split this method up into three methods:

1) `waitForAllocTerminal(allocID string) *structs.Allocation`
2) `retrieveRemoteAllocDir(alloc *structs.Allocation) *AllocDir`
3) 
```
func(c *Client) migrateRemoteAllocDir(alloc *structs.Allocation) {
   oldAlloc := waitForAllocTerminal(alloc.PreviousAllocation)
   oldAllocDir := retrieveRemoteAllocDIr(oldAlloc)
  c.addAlloc(alloc, oldAllocDir)
}
```